documentation/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.md

```diff
@@ -0,0 +1,224 @@
+      background: "var(--color-background)",
+    },
+  },
+}
+```
+
+In a component, we use the Tailwind classes:
+
+```vue
+<template>
+  <!-- This will be black in light mode and white in dark mode! -->
+  <p class="text-foreground">Hello World</p>
+</template>
+```
+
+Finally, if we needed an "escape hatch" to make sure this component was _always_
+black, _regardless_ of dark mode:
+
+```vue
+<template>
+  <p class="text-foreground dark:text-background">Hello World</p>
+</template>
+```
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+- Users writing components will have a "dark mode compatible by default"
+  experience. By default, they will not need to think much about dark mode.
+- Color names in the codebase will be replaced with semantic names.
+- Frontend developers will have easy tools to visually test components in light
+  and dark mode.
+
+## Rejected alternate approaches
+
+- Using the `dark:` modifier _exclusively_ for dark mode styling. This is more
+  explicit, but much, much more verbose, and would require extensive edits to
+  every single component we have written. Instead of writing `bg-background`,
+  for example, we would have to write `bg-white dark:bg-black` all throughout
+  the codebase.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+The following plan requires approved designs and semantic color names.
+
+1. Parallel Work Stream "A": Implement the new color palette
+   1. Create a `FORCE_DARK_MODE` feature flag which is disabled by default and
+      available in every environment. Add logic to add a `dark-mode` class to
+      the root HTML tag when this flag is enabled and a `light-mode` class when
+      disabled.
+   2. Rename all colors in the tailwind config and the frontend components to
+      use new, semantic names.
+   3. Replace "hardcoded" color values in the tailwind configuration file with
+      css variables defined in the "base" layer of the `tailwind.css` file.
+   4. Add the dark mode colors in the form of additional css variable
+      definitions, nested under the `.dark-mode` css class, in the "base" layer
+      of the `tailwind.css` file. **At this point, the full dark mode appearance
+      should be able to be tested manually by @fcoveram and
+      @wordpress/openverse-frontend for any inconsistiencies or problems**.
+   5. Create a `color-mode.ts` test utility for playwright that works comparably
```

Comment:
I think I would prefer that the initial screenshots are as correct as possible, rather than commit screenshots with errors only to be replaced in subsequent PRs. Can you think of any benefits not to initially commit the screenshots once the components look right in a PR?